common/primitives/core/src/omni/intent.rs

```diff
@@ -4,23 +4,30 @@ use scale_info::TypeInfo;
 use sp_core::H160;
 use sp_runtime::{traits::ConstU32, BoundedVec};
 
-pub const CALL_ETHEREUM_INPUT_LEN: u32 = 10 * 1024;
-pub const MAX_REMARK_LEN: u32 = u32::max_value();
+pub type CallEthereumInputLen = ConstU32<{ 10 * 1024 }>;
+pub type RemarkLen = ConstU32<{ u32::max_value() }>;
+pub type IntentStringLen = ConstU32<128>;
 
-#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo)]
+pub type IntentId = u32;
```

Comment:
Do you think we will have other types of intents that will require `intentId` ? We could use `SwapOrderId` (or something similar) and make it part of the `Swap` intent

Reply:
We have - theoretically all "intents" should have an `IntentId` 🤔  like ethereum contract call - as long as it needs to be tracked and order needs to be preserved.

Reply:
Well, in the ERC-7683 specs there is no intent id. But I can see the usefulness, depending on the usage